linera-sdk/src/contract/conversions_from_wit.rs

```diff
@@ -51,10 +51,7 @@ impl From<wit_system_api::Blob> for Blob {
 
 impl From<wit_system_api::HashedBlob> for HashedBlob {
     fn from(hashed_blob: wit_system_api::HashedBlob) -> Self {
-        HashedBlob {
-            id: hashed_blob.id.into(),
-            blob: hashed_blob.blob.into(),
-        }
+        HashedBlob::new(hashed_blob.id.into(), hashed_blob.blob.into())
```

Comment:
Why do we need this conversion? The host can't trust a contract to have computed the hash correctly.

Reply:
Ok, let me see if I can make it checked

Reply:
But why do we even need this conversion?

Reply:
For the `contract`/`service` runtime `read_blob` call

Reply:
Ah, I see, on the Wasm side! Maybe we should pass only the unhashed blob across the WIT boundary? Then `read_blob` implementation in the contract _runtime_ can then add the blob ID (unchecked), because it already knows which blob it asked for?